nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/encrypt/StringEncryptor.java

```diff
@@ -267,6 +296,16 @@ private boolean paramsAreValid() {
         return algorithmAndProviderValid && secretIsValid;
     }
 
+    private boolean customSecretIsValid(PBEKeySpec password, SecretKeySpec key, String algorithm) {
+        // Currently, the only custom algorithms use AES-G/CM with a password via Argon2
+        String rawPassword = new String(password.getPassword());
+        final boolean secretIsValid = StringUtils.isNotBlank(rawPassword) && rawPassword.trim().length() >= 12;
+        if (!secretIsValid) {
+            logger.error("The nifi.sensitive.props.key password provided is invalid for algorithm {}; must be >= 12 characters", algorithm);
```

Comment:
This message is really helpful for the user to know the problem. One minor issue is that the exception that prints to the log doesn't end up including this, it just ends with:
```

Caused by: org.apache.nifi.encrypt.EncryptionException: Cannot initialize the StringEncryptor because some configuration values are invalid
	at org.apache.nifi.encrypt.StringEncryptor.initialize(StringEncryptor.java:270)
	at org.apache.nifi.encrypt.StringEncryptor.<init>(StringEncryptor.java:137)
	at org.apache.nifi.encrypt.StringEncryptor.createEncryptor(StringEncryptor.java:236)
	at org.apache.nifi.encrypt.StringEncryptor.createEncryptor(StringEncryptor.java:209)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	... 58 common frames omitted
```

Was wondering if we could get that same message into the exception so its not missed and is clear what needs to be fixed.

Reply:
Just to clarify, we do print this first....
```

2020-07-24 17:12:31,384 ERROR [main] org.apache.nifi.encrypt.StringEncryptor ********************************************************************************
2020-07-24 17:12:31,385 ERROR [main] org.apache.nifi.encrypt.StringEncryptor *                A blank sensitive properties key was provided                 *
2020-07-24 17:12:31,385 ERROR [main] org.apache.nifi.encrypt.StringEncryptor *                   Specify a unique key in nifi.properties                    *
2020-07-24 17:12:31,385 ERROR [main] org.apache.nifi.encrypt.StringEncryptor *                         for nifi.sensitive.props.key                         *
2020-07-24 17:12:31,385 ERROR [main] org.apache.nifi.encrypt.StringEncryptor *                                                                              *
2020-07-24 17:12:31,385 ERROR [main] org.apache.nifi.encrypt.StringEncryptor *            The Encrypt Config Tool in NiFi Toolkit can be used to            *
2020-07-24 17:12:31,385 ERROR [main] org.apache.nifi.encrypt.StringEncryptor *                       migrate the flow to the new key                        *
2020-07-24 17:12:31,385 ERROR [main] org.apache.nifi.encrypt.StringEncryptor ********************************************************************************
2020-07-24 17:12:31,385 ERROR [main] org.apache.nifi.encrypt.StringEncryptor The nifi.sensitive.props.key password provided is invalid for algorithm NIFI_ARGON2_AES_GCM_256; must be >= 12 characters
```
but then after there are two stacktraces that I think overshadow this and don't have any specifics.

Reply:
Great point Bryan. Fixed.